### PR TITLE
[CMYK] Properties widget

### PR DESCRIPTION
**Pull request by troopa81**
Add CMYK project properties configuration widgets.

When loading and ICC profile, QGIS force color model to keep consistency between color model and color space.

**ICC profile widget part is not visible if QGIS is not built with Qt 6.8.0 or greater** (That maybe require to be clearly advertise it t

cmyk_widget
he changelog)

**Funded by Bordeaux Métropole**

### CODE REVIEW

File: src/app/qgsprojectproperties.cpp [hunk 1]
```diff
@@ -2680,6 +2700,59 @@ void QgsProjectProperties::removeStyleDatabase()
...
 
+#if QT_VERSION >= QT_VERSION_CHECK(6, 8, 0)
+
+void QgsProjectProperties::addIccProfile()
+{
+  const QString iccProfileFilePath = QFileDialog::getOpenFileName(
+                                       this,
+                                       tr( "Load ICC Profile" ),
+                                       QDir::homePath(),
+                                       tr( "Style databases" ) + tr( "ICC Profile" ) + QStringLiteral( " (*.icc)" ) );
```
Comment (nyalldawson) [inline src/app/qgsprojectproperties.cpp:2711]:
```suggestion
                                       tr( "ICC Profile" ) + QStringLiteral( " (*.icc)" ) );
```

File: src/app/qgsprojectproperties.cpp [hunk 2]
```diff
@@ -2680,6 +2700,59 @@ void QgsProjectProperties::removeStyleDatabase()
...
+  updateColorSpaceWidget();
+}
+
+void QgsProjectProperties::removeIccProfile()
+{
+  mColorSpace = QColorSpace();
+  updateColorSpaceWidget();
+}
+
+void QgsProjectProperties::updateColorSpaceWidget()
```
Comment (nyalldawson) [inline src/app/qgsprojectproperties.cpp:2739]:
```suggestion
void QgsProjectProperties::updateColorSpaceWidgets()
```

File: src/app/qgsprojectproperties.h [hunk 3]
```diff
@@ -204,6 +205,34 @@ class APP_EXPORT QgsProjectProperties : public QgsOptionsDialogBase, private Ui:
     void removeStyleDatabase();
     void newStyleDatabase();
 
+#if QT_VERSION >= QT_VERSION_CHECK(6, 8, 0)
+
+    /**
+     * Called whenever user select the add ICC profile button
+     * \since QGIS 3.40
```
Comment (nyalldawson) [inline src/app/qgsprojectproperties.h:212]: We don't need these for app classes: (they aren't exposed anywhere)

```suggestion
```

**Comment by nyalldawson** [comment]
Code looks good!

I have two thoughts:

1. I think the "style" tab is becoming a random mash of settings now. I think we should probably pull these new settings and the existing "Project Colors" list out into a new dedicated "Colors" tab. 
2. Do you think we should add a button to export the current profile to a file on disk? If eg a user shares a project with someone else, they may want to easily extract this profile to use for another project or in another application.

**Comment by troopa81** [comment]
> I have two thoughts:

Good points. Done. Screenshot has been updated

File: src/app/qgsprojectproperties.cpp [hunk 4]
```diff
@@ -2680,6 +2702,76 @@ void QgsProjectProperties::removeStyleDatabase()
...
+
+void QgsProjectProperties::removeIccProfile()
+{
+  mColorSpace = QColorSpace();
+  updateColorSpaceWidgets();
+}
+
+void QgsProjectProperties::saveIccProfile()
+{
+  QString fileName = QFileDialog::getSaveFileName( this, tr( "Save ICC profile" ), QDir::homePath(),
```
Comment (nyalldawson) [inline src/app/qgsprojectproperties.cpp:2743]:
```suggestion
  QString fileName = QFileDialog::getSaveFileName( this, tr( "Save ICC Profile" ), QDir::homePath(),
```

File: src/app/qgsprojectproperties.cpp [hunk 5]
```diff
@@ -2680,6 +2702,76 @@ void QgsProjectProperties::removeStyleDatabase()
...
+                     tr( "ICC profile files (*.icc *.ICC)" ) );
+
+  if ( fileName.isEmpty() )
+    return;
+
+  const QString error = QgsColorUtils::saveIccProfile( mColorSpace, fileName );
```
Comment (nyalldawson) [inline src/app/qgsprojectproperties.cpp:2749]: We need this to ensure the extension is always present on all environments (eg gnome won't add it by default):

```suggestion
  fileName = QgsFileUtils::ensureFileNameHasExtension( fileName, { QStringLiteral( "icc" ) } );
  const QString error = QgsColorUtils::saveIccProfile( mColorSpace, fileName );
```

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**